pkg/kube/client/client.go

```diff
@@ -103,6 +104,10 @@ func (c client) GetSecret(objectKey k8sClient.ObjectKey) (corev1.Secret, error)
 	return s, nil
 }
 
+func (c client) IsNotFound(err error) bool {
```

Comment:
Q: Why is this wrapping needed?

Reply:
it is not exactly needed, but it allows us to pass a secret client which implements the same without having to duplicate the custom `NotFound` code in a lot of places

Reply:
hmm, I merged a change that checks for secretNotFound -- basically the same logic as enterprise -- since we need it for scram secret creation. Do you think we can re-use it?

Reply:
sure, I'll do it